utility/pbkdf2-hash-generator/src/main/java/org/wso2/carbon/identity/utility/hash/generator/pbkdf2/PBKDF2HashGenerator.java

```diff
@@ -0,0 +1,128 @@
+import java.util.Base64;
+import java.util.Scanner;
+
+/**
+ * This class contains the Client for generate PBKDF2 Hash and the Salt Value for a given plain text password.
+ */
+public class PBKDF2HashGenerator {
+
+    private static final Log log = LogFactory.getLog(PBKDF2HashGenerator.class);
+    private static final String SHA_1_PRNG = "SHA1PRNG";
+    private static PBKDF2HashProvider pbkdf2HashProvider;
+
+    public static void main(String[] args) {
+
+        log.info("---Started the Hash generation Script---");
+        initHashProvider();
+        getPasswordFromScanner();
+        log.info("---Hash generation Task Completed---");
+    }
+
+    /**
+     * Get Passwords as a user input from cmd line and print hashed password and salt value.
+     */
+    private static void getPasswordFromScanner() {
+
+        Scanner scanner = new Scanner(System.in, "UTF8");
+
+        while (true) {
+
+            log.info("Enter Password:");
+            //reads password.
+            String originalPassword = scanner.nextLine();
+
+            String generatedSalt = generateSaltValue();
+            byte[] generatedSecuredPasswordHash;
+            try {
+                generatedSecuredPasswordHash = calculateHash(originalPassword.toCharArray(), generatedSalt);
+            } catch (HashProviderException e) {
+                log.error(e);
+                break;
+            }
+            String strongPassword = bytesToBase64(generatedSecuredPasswordHash);
+
+            log.info("Hashed Password:" + strongPassword);
+            log.info("Salt Value:" + generatedSalt);
+            log.info("----------------------------------");
+        }
+    }
+
+    /**
+     * This method used to initialize the PBKDF2HashProvider instance.
+     */
+    private static void initHashProvider() {
+
+        pbkdf2HashProvider = new PBKDF2HashProvider();
+        pbkdf2HashProvider.init();
+    }
+
+    /**
+     * Generate hash password for the given plain password and salt value.
+     *
+     * @param plainText The plain text value to be hashed.
+     * @param salt      The salt value.
+     * @return The resulting hash value of the value.
+     */
+    private static byte[] calculateHash(char[] plainText, String salt) throws HashProviderException {
+
+        return pbkdf2HashProvider.calculateHash(plainText, salt);
+    }
+
+    /**
+     * This private method returns a saltValue using SecureRandom.
+     *
+     * @return saltValue.
+     */
+    private static String generateSaltValue() throws RuntimeException {
+
+        try {
+            SecureRandom secureRandom = SecureRandom.getInstance(SHA_1_PRNG);
+            byte[] bytes = new byte[16];
+            //secureRandom is automatically seeded by calling nextBytes
+            secureRandom.nextBytes(bytes);
+            return bytesToBase64(bytes);
+        } catch (NoSuchAlgorithmException e) {
+            throw new RuntimeException("SHA1PRNG algorithm could not be found.");
```

Comment:
its not recommended to throw RuntimeExceptions unless necessary. Shall we throw a checked exception from here and handle it properly where this method is getting called.

Reply:
Addressed with https://github.com/wso2-extensions/identity-hash-provider-pbkdf2/pull/12